File: JavaScript/5kyu/GreedIsGood.js
function score(dice) {
  var tripple = [0, 1000, 200, 300, 400, 500, 600];
  var other = [0, 100, 0, 0, 0, 50, 0];

  var result = 0;

  for (var diceSide = 1; diceSide <= 6; diceSide++) {
    var rolls = dice.filter((r) => r == diceSide).reduce((a, v) => (a += 1), 0);
    result +=
      tripple[diceSide] * parseInt(rolls / 3) + other[diceSide] * (rolls % 3);
  }

  return result;
}

// score([2, 3, 4, 6, 2]);

export { score };

File: Python/5kyu/GreedIsGood.py
def score(dice):
    tripple = [0, 1000, 200, 300, 400, 500, 600]
    other = [0, 100, 0, 0, 0, 50, 0]
    result = 0

    for side in range(1, 7):
        rolls = sum([1 for d in dice if d == side])
        result += tripple[side] * int(rolls / 3) + other[side] * (rolls % 3)

    return result


def score_kata_1(dice):
    sum = 0
    counter = [0, 0, 0, 0, 0, 0]
    points = [1000, 200, 300, 400, 500, 600]
    extra = [100, 0, 0, 0, 50, 0]

    for die in dice:
        counter[die-1] += 1

    for (i, count) in enumerate(counter):
        sum += points[i] if count >= 3 else 0
        sum += extra[i] * (count % 3)


print(score([4, 4, 4, 3, 3]))
